lib/services/search_service.dart

```diff
@@ -1,14 +1,35 @@
-import 'package:concordia_go/models/concordia_building_model.dart';
+import 'package:concordia_go/models/classroom.dart';
+import 'package:concordia_go/models/concordia_building.dart';
+import 'package:concordia_go/models/direction_object.dart';
 import 'package:concordia_go/utilities/concordia_constants.dart';
 
 class SearchService {
-  static List<ConcordiaBuilding> getSearchResults(String query) {
-    var searchResults = <ConcordiaBuilding>[];
+  SearchService._privateConstructor();
 
-    for (var value in buildings.values) {
-      if (value.name.toLowerCase().contains(query.toLowerCase()) ||
-          value.code.toLowerCase().contains(query.toLowerCase())) {
-        searchResults.add(value);
+  static final SearchService instance = SearchService._privateConstructor();
+
+  List<Dobject> getSearchResults(String query) {
+    final List<Dobject> searchResults = <Dobject>[];
+
+    for (final ConcordiaBuilding building in buildings.values) {
+      if (building.name.toLowerCase().contains(query.trim().toLowerCase()) ||
+          building.code.toLowerCase().contains(query.trim().toLowerCase())) {
+        searchResults.add(Dobject.building(building));
+      }
+    }
+
+    if (query.isNotEmpty) {
+      for (final Classroom room in rooms) {
+        String name;
+        if (room.building.code == 'MB') {
+          name = room.building.code + room.floor + '.' + room.number;
+        } else {
+          name = room.building.code + room.number;
+        }
```

Comment:
Is this special case only for MB buildings? If not, that could cause us to have to come back every time and modify this file with many cases of "if" statements which isn't really good practice.

Reply:
@Debnam @MoreWaqar